services/discourse/discourse.service.js

```diff
@@ -1,176 +1,110 @@
-        namedParams: {
-          scheme: 'https',
-          host: 'meta.discourse.org',
-        },
-        staticPreview: {
-          label: 'discourse',
-          message: 'online',
-          color: 'brightgreen',
-        },
+        staticPreview: this.render(),
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/discourse\/(http(?:s)?)\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const scheme = match[1] // eg, https
-        const host = match[2] // eg, meta.discourse.org
-        const stat = match[3] // eg, user_count
-        const format = match[4]
-        const url = `${scheme}://${host}/site/statistics.json`
-
-        const options = {
-          method: 'GET',
-          uri: url,
-          headers: {
-            Accept: 'application/json',
-          },
-        }
-
-        const badgeData = getBadgeData('discourse', data)
-        request(options, (err, res) => {
-          if (err != null) {
-            if (res) {
-              console.error(`${res}`)
-            }
-
-            badgeData.text[1] = 'inaccessible'
-            sendBadge(format, badgeData)
-            return
-          }
-
-          if (res.statusCode !== 200) {
-            badgeData.text[1] = 'inaccessible'
-            badgeData.colorscheme = 'red'
-            sendBadge(format, badgeData)
-            return
-          }
-
-          badgeData.colorscheme = 'brightgreen'
-
-          try {
-            const data = JSON.parse(res['body'])
-            let statCount
-
-            switch (stat) {
-              case 'topics':
-                statCount = data.topic_count
-                badgeData.text[1] = `${metric(statCount)} topics`
-                break
-              case 'posts':
-                statCount = data.post_count
-                badgeData.text[1] = `${metric(statCount)} posts`
-                break
-              case 'users':
-                statCount = data.user_count
-                badgeData.text[1] = `${metric(statCount)} users`
-                break
-              case 'likes':
-                statCount = data.like_count
-                badgeData.text[1] = `${metric(statCount)} likes`
-                break
-              case 'status':
-                badgeData.text[1] = 'online'
-                break
-              default:
-                badgeData.text[1] = 'invalid'
-                badgeData.colorscheme = 'yellow'
-                break
-            }
-
-            sendBadge(format, badgeData)
-          } catch (e) {
-            console.error(`${e.stack}`)
-            badgeData.colorscheme = 'yellow'
-            badgeData.text[1] = 'invalid'
-            sendBadge(format, badgeData)
-          }
-        })
-      })
-    )
+  static render() {
+    return {
+      message: 'online',
+      color: 'brightgreen',
+    }
+  }
+
+  async handle({ scheme, host }) {
```

Comment:
I guess what’s happening here is that if the fetch fails, we are offline. It’s a little magical; could you add a comment explaining that?